src/lib/form-field/form-field.ts

```diff
@@ -448,6 +452,13 @@ export class MatFormField extends _MatFormFieldMixinBase
       return;
     }
 
+    // If the element is not present in the DOM, the outline gap will need to be calculated
+    // the next time it is checked and in the DOM.
+    if (document && !document.documentElement.contains(this._elementRef.nativeElement)) {
```

Comment:
use `platform.isBrowser` https://github.com/angular/material2/blob/master/src/cdk/platform/platform.ts#L29 (I see there's already a check for this down below, maybe you can simplify the logic a bit)